spirv_msl.cpp

```diff
@@ -1004,13 +1137,15 @@ void CompilerMSL::add_plain_variable_to_interface_block(StorageClass storage, co
 		set_member_decoration(ib_type.self, ib_mbr_idx, DecorationCentroid);
 	if (is_sample)
 		set_member_decoration(ib_type.self, ib_mbr_idx, DecorationSample);
+
+	set_extended_member_decoration(ib_type.self, ib_mbr_idx, SPIRVCrossDecorationInterfaceOrigID, var.self);
```

Comment:
What is this OrigID used for? I cannot find any "get" of it.

Reply:
It's used after the input struct is sorted, to map the member back to its original variable. We need to do this in order to set the member index, so we can get access chains right.

Reply:
[Here](https://github.com/KhronosGroup/SPIRV-Cross/pull/852/files#diff-6a4029d3c45d50818e6f80e915867c64R1614). I used it directly since I have the meta, instead of going through the function.

Reply:
Oh, right.